pkg/servicemesh/federation/common/util.go

```diff
@@ -94,3 +101,20 @@ func RemoteChecksum(remote v1.ServiceMeshPeerRemote) uint64 {
 	}
 	return checksum
 }
+
+// hashResourceName applies a sha256 on the host and truncates it to the first n char
+func hashResourceName(name string, n int) string {
+	hash := sha256.Sum256([]byte(name))
+	return (hex.EncodeToString(hash[:]))[:n]
```

Comment:
Is the `[:]` necessary?

Reply:
It is the way for converting the `[32]byte` result from the `Sum256()` to the `[]byte` required by the `EncodeToString()`